.github/workflows/build.yaml

```
@@ -45,3 +45,13 @@ jobs:
       branch: ${{ inputs.branch }}
       date: ${{ inputs.date }}
       sha: ${{ inputs.sha }}
+  docs-build:
+    if: ${{ startsWith(github.ref, 'refs/heads/branch-') }}
```

```suggestion
    if: github.ref_type == 'branch' && github.event_name == 'push'
```

I just discovered this problem that we'll need to update in all of the docs build PRs that have already been merged.

We need to change this conditional so that this job doesn't run as part of the nightly builds.

The docs builds aren't intended to run as part of the nightly jobs, so they're currently breaking the nightly builds (see cuml nightly job [here](https://github.com/rapidsai/cuml/actions/runs/4171191106) for reference).

We can update this conditional as shown here to:

- simplify the `branch` assertion
- ensure that it only runs on `push` events (as opposed to `workflow_dispatch` events which trigger the nightlies).

After this PR is merged, we can test that this conditional works correctly and then roll out the fix to other repositories.

cc: @AjayThorve